core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/PathHandler.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.store.operation.handler;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import uk.gov.gchq.gaffer.commonutil.stream.Streams;
+import uk.gov.gchq.gaffer.data.element.Edge;
+import uk.gov.gchq.gaffer.data.element.Element;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.data.EntitySeed;
+import uk.gov.gchq.gaffer.operation.impl.Path;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.operation.impl.output.ToEntitySeeds;
+import uk.gov.gchq.gaffer.operation.impl.output.ToSet;
+import uk.gov.gchq.gaffer.operation.impl.output.ToVertices;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Stack;
+import java.util.stream.Collectors;
+
+public class PathHandler implements OutputOperationHandler<Path, Iterable<Iterable<Edge>>> {
+
+    private final Set<EntitySeed> visitedSeeds = new HashSet<>();
+    private int pathVertices;
+    private int hops;
+
+    @Override
+    public Iterable<Iterable<Edge>> doOperation(final Path operation, final Context context, final Store store) throws OperationException {
+
+        // Check input
+        if (null == operation.getInput() || !operation.getInput().iterator().hasNext()) {
```

Comment:
We really need to avoid doing operation.getInput().iterator().hasNext(). It is really inefficient for when run on results from accumulo and it causes the connection to accumulo to not be closed properly.

Reply:
If the input is empty then I think we should return an empty iterable to make it consistent with other operations.